libs/rtemodel/src/RtePackage.cpp

```diff
@@ -743,6 +743,26 @@ string RtePackage::ConstructID()
 }
 
 
+string RtePackage::GetPackYamlID() const
```

Comment:
Couldn't `string ProjMgrUtils::GetPackageID(const RteItem* pack)` be used instead of introducing this function?

Reply:
Not sure if we want to introduce the YAML format awareness into the `RtePackage` or if it's better to keep it limited to the `projmgr` tool.
As previously commented the variables expansion should be done via callbacks which can be tool's specific.
@edriouk can you please advise?

Reply:
We are already considering to unify component, API, and pack IDs across tools and libraries. RTE model will use then format used in .cproject/.csolution.   This happens not for 2.0 release, but soon. therefore I agree with @brondani  not to introduce extra methods in RTE model.

Reply:
Do you have a way to fetch the generator of interest from the callback class?
As far as I can tell, only global context is available and no way to get the _current_ generator.